index.js

```diff
@@ -11,17 +11,16 @@ var resolve = require('resolve')
 var getPackageInfo = require('get-package-info')
 var common = require('./common')
 
-var supportedArchs = {
-  ia32: 1,
-  x64: 1
-}
-
-var supportedPlatforms = {
-  // Maps to module ID for each platform (lazy-required if used)
-  darwin: './mac',
-  linux: './linux',
-  win32: './win32'
-}
+var supportedArchs = common.archs.reduce(function (o, v) {
+  o[v] = 1
+  return o
+}, {})
+
+// Maps to module ID for each platform (lazy-required if used)
+var supportedPlatforms = common.platforms.reduce(function (o, v) {
+  o[v] = './' + (v === 'darwin' ? 'mac' : v)
+  return o
+}, {})
```

Comment:
Variables renamed. I don't want to pollute file scope, but cannot use `let`, so, still use `reduce` instead of `for in` (`for of` es6).

Mac App Store (MAS) platform can be added as before — directly to supportedPlatforms (or inside reduce function — in any case we check `is equal to darwin?`).